contracts/Roles.sol

```diff
@@ -126,126 +124,165 @@ contract Roles is Modifier {
+    /// @param allow Bool to allow (true) or disallow (false) call to an address.
+    function allowTarget(
         uint16 role,
         address targetAddress,
-        bool scoped
+        bool allow
     ) external onlyOwner {
-        roleList.roles[role].targets[targetAddress].clearance = scoped
-            ? Clearance.FUNCTION
+        roles[role].targets[targetAddress].clearance = allow
+            ? Clearance.TARGET
             : Clearance.NONE;
-        emit SetTargetAddressScoped(role, targetAddress, scoped);
+
+        emit AllowTarget(role, targetAddress, allow);
+    }
+
+    /// @dev Allows a specific function on a target address to be called.
+    /// @notice Only callable by owner.
+    /// @param role Role to set for
+    /// @param targetAddress Scoped address on which a function signature should be allowed/disallowed.
+    /// @param functionSig Function signature to be allowed/disallowed.
+    /// @param allow Bool to allow (true) or disallow (false) call to a function on an address.
+    function allowFunction(
+        uint16 role,
+        address targetAddress,
+        bytes4 functionSig,
+        bool allow
+    ) external onlyOwner {
+        if (allow) {
+            roles[role].targets[targetAddress].clearance = Clearance.FUNCTION;
+        }
+
+        roles[role].functions[
+            Permissions.keyForFunctions(targetAddress, functionSig)
+        ] = allow ? Permissions.FUNCTION_WHITELIST : 0;
+        emit AllowFunction(role, targetAddress, functionSig, allow);
     }
 
-    /// @dev Sets whether or not calls should be scoped to specific parameter value or range of values.
+    /// @dev Sets and enforces scoping for a function on an address
     /// @notice Only callable by owner.
     /// @param role Role to set for.
     /// @param targetAddress Address to be scoped/unscoped.
     /// @param functionSig first 4 bytes of the sha256 of the function signature.
-    /// @param paramsScoped false for un-scoped, true for scoped.
-    /// @param types false for static, true for dynamic.
-    /// @param compTypes Any, or EqualTo, GreaterThan, or LessThan compValue.
-    function setParametersScoped(
+    /// @param isParamScoped false for un-scoped, true for scoped.
+    /// @param isParamDynamic false for static, true for dynamic.
+    /// @param paramCompType Any, or EqualTo, GreaterThan, or LessThan compValue.
+    function scopeFunction(
         uint16 role,
         address targetAddress,
         bytes4 functionSig,
-        bool[] memory paramsScoped,
-        bool[] memory types,
-        Comp.Comparison[] memory compTypes
+        bool[] calldata isParamScoped,
+        bool[] calldata isParamDynamic,
+        Comparison[] calldata paramCompType
     ) external onlyOwner {
-        Permissions.setParametersScoped(
-            roleList,
-            role,
-            targetAddress,
-            functionSig,
-            paramsScoped,
-            types,
-            compTypes
+        // 24kb
```

Comment:
Did you mean to leave this chunk of commented out code?

Reply:
Left it, since this was not a final branch. Will push the removal.